components/core/src/clp/hash_utils.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef CLP_HASH_UTILS_HPP
+#define CLP_HASH_UTILS_HPP
+
+#include <cstddef>
+#include <span>
+#include <string>
+#include <vector>
+
+#include <openssl/evp.h>
+
+#include "ErrorCode.hpp"
+#include "TraceableException.hpp"
+
+namespace clp {
+/**
+ * A C++ wrapper for openssl's EVP digest message digest context (EVP_MD_CTX).
+ */
+class EvpDigestContext {
+public:
+    // Types
+    class OperationFailed : public TraceableException {
+    public:
+        // Constructors
+        OperationFailed(ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+
+        // Methods
+        [[nodiscard]] auto what() const noexcept -> char const* override {
+            return "EvpCtxManager operation failed";
+        }
+    };
+
+    // Constructors
+    EvpDigestContext(const EVP_MD* type, ENGINE* impl) : m_md_ctx{EVP_MD_CTX_create()} {
```

Comment:
```suggestion
    EvpDigestContext(EVP_MD const* type, ENGINE* impl) : m_md_ctx{EVP_MD_CTX_create()} {
```
It's annoying that `clang-format` doesn't catch this one...